pkg/controller/customresource/customresource.go

```diff
@@ -139,3 +140,34 @@ func SetAnnotation(resource api.AtlasCustomResource, key, value string) {
 	annot[key] = value
 	resource.SetAnnotations(annot)
 }
+
+func ComputeSecretWithFallback[R api.ResourceWithCredentials](fallback bool, project *akov2.AtlasProject, resource R) (*client.ObjectKey, error) {
+	secret, err := resolveConnectionSecret(project, resource)
+	if fallback {
+		return secret, nil
+	}
+	return secret, err
+}
+
+func resolveConnectionSecret[R api.ResourceWithCredentials](project *akov2.AtlasProject, resource R) (*client.ObjectKey, error) {
+	if reflect.ValueOf(&resource).Elem().IsZero() {
```

Comment:
let me think how we can simplify this. I don't see reflection is necessary here.

Reply:
Fixed